autosubmit/job/job_list.py

```diff
@@ -3049,7 +2977,7 @@ def print_with_status(self, status_change: Optional[dict[Any, Any]] = None, noco
         for job in all_jobs:
             if len(job.parents) == 0:
                 roots.append(job)
-        visited = list()
+        visited: list = []
```

Comment:
when doing it like this, it can be useful to specify the list element (e.g. `list[str]`) when we know it. Will check what's the type here later.